shared/src/main/scala/amf/core/remote/Context.scala

```diff
@@ -1,93 +1,102 @@
   */
-class Context protected (val platform: Platform,
-                         val history: List[String],
-                         val mappings: Map[String, String] = Map.empty) {
+class Context protected(val platform: Platform,
+                        val history: List[String]) {
 
   def hasCycles: Boolean = history.count(_.equals(current)) == 2
 
   def current: String = if (history.isEmpty) "" else history.last
   def root: String    = if (history.isEmpty) "" else history.head
 
-  def update(url: String): Context = Context(platform, history, resolve(url), mappings)
+  def update(url: String): Context = Context(platform, history, resolve(url))
 
   def resolve(url: String): String =
     try {
-      applyMapping(platform.resolvePath(resolvePathAccordingToRelativeness(url)))
+      resolvePathAccordingToRelativeness(url)
     } catch {
       case e: Exception => throw new PathResolutionError(s"url: $url - ${e.getMessage}")
     }
 
   private def resolvePathAccordingToRelativeness(url: String) = {
-    val prefix = url match {
-      case Absolute(_)               => ""
-      case RelativeToRoot(_)         => Context.stripFile(root, platform.operativeSystem())
-      case RelativeToIncludedFile(_) => Context.stripFile(current, platform.operativeSystem())
+    val base = url match {
+      case Absolute(_)               => None
+      case RelativeToRoot(_)         => Some(root)
+      case RelativeToIncludedFile(_) => Some(current)
     }
-    concatWithoutDuplicateSlash(prefix, url)
-  }
-
-  private def concatWithoutDuplicateSlash(prefix: String, url: String) = {
-    if (prefix.nonEmpty && prefix.last == '/' && url.nonEmpty && url.head == '/') prefix + url.drop(1)
-    else prefix + url
+    resolveRelativeTo(base, url)
   }
 
-  private def applyMapping(path: String): String =
-    mappings.find(m => path.startsWith(m._1)).fold(path)(m => path.replace(m._1, m._2))
 }
 
 object Context {
   private def apply(platform: Platform,
                     history: List[String],
-                    currentResolved: String,
-                    mapping: Map[String, String]): Context =
-    new Context(platform, history :+ currentResolved, mapping)
-
-  def apply(platform: Platform, root: String): Context = Context(platform, root, Map.empty)
+                    currentResolved: String): Context =
+    new Context(platform, history :+ currentResolved)
 
-  def apply(platform: Platform): Context = Context(platform, "", Map.empty)
+  def apply(platform: Platform): Context = new Context(platform, Nil)
 
-  def apply(platform: Platform, root: String, mapping: Map[String, String]): Context =
-    new Context(platform, Option(root).filter(_.nonEmpty).map(List(_)).getOrElse(Nil), mapping)
+  def apply(platform: Platform, root: String): Context =
+    new Context(platform, List(root))
 
-  def apply(platform: Platform, mapping: Map[String, String]): Context =
-    new Context(platform, Nil, mapping)
+  def stripFileName(url: String): String = {
```

Comment:
This should be extracted in an object and unit tested separately. It would be great if we had a test for each OS also.